src/main/java/org/skriptlang/reflect/syntax/condition/elements/StructCustomCondition.java

```diff
@@ -0,0 +1,166 @@
+import com.btk5h.skriptmirror.util.SkriptUtil;
+import org.skriptlang.reflect.syntax.condition.ConditionCheckEvent;
+import org.skriptlang.reflect.syntax.condition.ConditionSyntaxInfo;
+import org.skriptlang.skript.lang.entry.EntryContainer;
+import org.skriptlang.skript.lang.script.Script;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class StructCustomCondition extends CustomSyntaxStructure<ConditionSyntaxInfo> {
+
+  public static boolean customConditionsUsed = false;
+
+  static {
+    String[] syntax = {
+        "[:local] condition <.+>",
+        "[:local] condition",
+        "[:local] %*classinfos% property condition <.+>"
+    };
+    Skript.registerStructure(StructCustomCondition.class, customSyntaxValidator()
+        .addSection("check", false)
+        .build(), syntax);
+  }
+
+  private static final DataTracker<ConditionSyntaxInfo> dataTracker = new DataTracker<>();
+
+  static final Map<ConditionSyntaxInfo, Trigger> conditionHandlers = new HashMap<>();
+  static final Map<ConditionSyntaxInfo, Trigger> parserHandlers = new HashMap<>();
+  static final Map<ConditionSyntaxInfo, List<Supplier<Boolean>>> usableSuppliers = new HashMap<>();
+  static final Map<ConditionSyntaxInfo, Boolean> parseSectionLoaded = new HashMap<>();
+
+  static {
+    Skript.registerCondition(CustomCondition.class);
+    Optional<SyntaxElementInfo<? extends Condition>> info = Skript.getConditions().stream()
+        .filter(i -> i.c == CustomCondition.class)
+        .findFirst();
+    info.ifPresent(dataTracker::setInfo);
+
+    dataTracker.addManaged(conditionHandlers);
+    dataTracker.addManaged(parserHandlers);
+    dataTracker.addManaged(usableSuppliers);
+    dataTracker.addManaged(parseSectionLoaded);
+  }
+
+  private SectionNode parseNode;
+
+  @Override
+  public DataTracker<ConditionSyntaxInfo> getDataTracker() {
+    return dataTracker;
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public boolean init(Literal<?>[] args, int matchedPattern, SkriptParser.ParseResult parseResult,
+                      EntryContainer entryContainer) {
+    customConditionsUsed = true;
+
+    List<String> patterns = entryContainer.getOptional("patterns", List.class, false);
+    Script script = parseResult.hasTag("local") ? SkriptUtil.getCurrentScript() : null;
+
+    if (matchedPattern != 1 && patterns != null) {
+      Skript.error("A custom condition with an inline pattern cannot have a 'patterns' entry too");
+      return false;
+    }
+
+    switch (matchedPattern) {
+      case 0: // condition with an inline pattern
+        String pattern = parseResult.regexes.get(0).group();
+        register(ConditionSyntaxInfo.create(script, pattern, 1, false, false));
+        break;
+      case 1: // condition with a 'patterns' entry
+        if (patterns == null) {
```

Comment:
Isn't this what the checkHasPatterns() is for?

Reply:
This is for a more specific error

Reply:
Is there a purpose for the checkHasPatterns method then?

Reply:
For a more general use case. For example if the `patterns` entry is present but empty.